### DISCUSSION TITLE
Ribbon view and residue-specific display

### DISCUSSION THREAD

**Discussion by aflaus** [Androp0v/BioViewer#14] [Ideas]
Excellent app thanks. Very big potential.

Single biggest enhancement needed is ribbon view (alpha helices, beta strands). Biomolecule structures are _way too big_ to view intelligibly atom-wise.

With this it would become useful to view structures personally (eg in audience or on train) or by projecting screen during presentation (eg teaching or symposium).

Second biggest need is residue-specific atom display (and colouring) on the ribbon backbone.

This would allow context of features of interest to be isolated and investigated. The interface to do this visually is an interesting question (eg via sequence panel). It would also need a way to save the selection/view.

Happy to help. Please be encouraged.

**Comment by Androp0v** [comment]
Thanks for the feedback! Really appreciate it. 

I agree that ɑ-helix/β-sheet visualization is the single biggest missing feature. I've been meaning to add that since forever, but I want to do it right from the start and that requires a couple extra features to be implemented beforehand (basically, straight helices/sheets are easy to do using just the annotations in the PDB file, but in order to have properly curved helices/sheets the program must detect the carbons in the backbone and fit the helix/sheet through them. So I'll have to implement backbone detection first at the very least, and then fit the helix/sheet trough the backbone of the structures).

As for residue selection, I prototyped a few different UIs for it a couple months ago, and it's the logical next step. Either that or a 'licorice' visualization mode, which is also on my TODO list.

As I said, thanks for the feedback. Glad to hear you're liking the app so far. I'm targeting mid-February for a ɑ-helix/β-sheet visualization mode, I'll update this thread with any news.

**Reply by aflaus** [reply]
Maybe this is a simplistic point you are already well aware of, but typically the backbone path is approximated by smooth interpolation through the "CA" atom name items in columns 13-16 of ATOM rows of a PDB file. The other covalently connected atom names for the backbone are "N" and "C".

The HELIX and SHEET rows just define the start and end points of sequential residues with backbone atom arrangements that satisfy appropriate stereochemistry for the respective  secondary structures. Apologies if you are aware of all this though.

**Reply by Androp0v** [reply]
Hey, just to keep you in the loop: that was actually super useful! I hadn't noticed that PDB files already kept track of the alpha carbons. Unfortunately, the class I created to parse PDB files back when I started the project wasn't very flexible, so in order to parse the alpha carbons and HELIX/SHEET records I had to rewrite the entire parser class to make it cleaner/more manageable.

That's mostly done now though (WIP is in open PR #15), so I can finally focus on writing the code to render the helices/sheets/loops. One nice thing about rewriting the parser is that it's now easier to keep track of per-atom properties, which allows the app to trivially add other coloring schemes (for example, the app can now color the atoms by residue type). The new parser is also multithreaded, so importing large files is much faster now.

Hopefully by the end of next week I should have some sort of crude version of the secondary structure visualization working, I'll keep this thread updated.

**Reply by aflaus** [reply]
Really good to hear thanks! Let me know if you want testing or feedback.

With respect to per-atom display like atoms and colouring, this gets complex so you might want to implement some sort of persistent user-defined preset recording capability.

**Reply by aflaus** [reply]
Hi. I was just wondering if you were able to make any progress with ribbon-cartoon views? Thanks.

**Comment by Guljenn** [comment]
I believe you guys have the wrong person
